### PR TITLE
Allows rotating computer frames, fixes rotation on deconstruction, switches dropped materials to use drop_location()

### LINKED ISSUE #32870: Computers rotate when screwdrivered
Probably also happens when you're building a new one off a frame.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #32870

🆑 ShizCalev
fix: Computers will no longer rotate incorrectly when being deconstructed.
fix: You can now rotate computer frames during construction.
/🆑

### CODE REVIEW

**Comment by ACCount12** [comment]
Beat me to it. 👍

File: code/game/machinery/computer/buildandrepair.dm [hunk 1]
```diff
@@ -138,8 +139,28 @@
...
+	
+	if(anchored)
+		to_chat(usr, "<span class='warning'>You must unwrench [src] before rotating it!</span>")
+		return
+
+	setDir(turn(dir, -90))
+
+/obj/structure/frame/computer/AltClick(mob/user)
+	..()
+	if(!in_range(src, user))
```
Comment (ChangelingRain) [inline code/game/machinery/computer/buildandrepair.dm]: this should be in rotate itself - also, this handles adjacency and ghosts, right?

Comment (ShizCalev) [reply code/game/machinery/computer/buildandrepair.dm]: It's pretty much a copy-paste of how all other AltClick()'s work.

**Comment by lzimann** [comment]
Conflicts.

**Comment by ShizCalev** [comment]
Welp fucked that up, will correct when I get home

File: code/game/machinery/computer/buildandrepair.dm [hunk 2]
```diff
@@ -138,9 +139,28 @@
...
-			new /obj/item/shard(loc)
-			new /obj/item/shard(loc)
+			new /obj/item/shard(drop_location())
+			new /obj/item/shard(drop_location())
 		if(state >= 3)
-			new /obj/item/stack/cable_coil(loc , 5)
+			new /obj/item/stack/cable_coil(drop_location(), 5)
+	..()
+
+/obj/structure/frame/computer/verb/rotate()
```
Comment (Jordie0608) [inline code/game/machinery/computer/buildandrepair.dm]: No new verbs please, move the functionality for this to AltClick (or wherever).

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/buildandrepair.dm [hunk 3]
```diff
@@ -138,9 +138,19 @@
...
+			new /obj/item/shard(drop_location())
+			new /obj/item/shard(drop_location())
 		if(state >= 3)
-			new /obj/item/stack/cable_coil(loc , 5)
+			new /obj/item/stack/cable_coil(drop_location(), 5)
+	..()
 
+/obj/structure/frame/computer/AltClick(mob/user)
 	..()
+	if(!in_range(src, user) || user.incapacitated())
```
Comment (AnturK) [inline code/game/machinery/computer/buildandrepair.dm]: Won't stop ghosts.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Pull request merged**